### PR TITLE
N-body simulation

### PR DESCRIPTION
**Pull request by samo-novak**
This adds an N-body simulation of the outer planets of the Solar system using a simple numerical integrator as required by the [benchmark problem description](https://benchmarksgame-team.pages.debian.net/benchmarksgame/description/nbody.html#nbody).

### TODO:

- [x] Simple implementation based on existing benchmarks
- [x] Check correctness of the simple version, numbers are ALMOST correct, but there is quite a large error currently:
  - [x] Check correctness of the sim (turns out I forgot to offset the sun's momentum at the start)
  - [x] Check correctness of energy computation
  - [x] Check if numbers entered are correct
- [x] This is a very stateful algorithm, hence implementation using handlers may be interesting
- [x] Headers, polishing

### CODE REVIEW

File: generic/nbody/handlers-mutation.links [hunk 1]
```diff
@@ -0,0 +1,389 @@
+# =================================================================
```
Comment (dhil) [inline generic/nbody/handlers-mutation.links]: I suggest create a directory `handlers/` in the root, and move this file there. Rename the file to `nbody.links`

File: generic/nbody/handlers-mutation-opt.links [hunk 2]
```diff
@@ -0,0 +1,385 @@
+# ====================================================================================
```
Comment (dhil) [inline generic/nbody/handlers-mutation-opt.links]: Move to `handlers/` and rename it `nbody_opt.links`

File: generic/nbody/simple.links [hunk 3]
```diff
@@ -0,0 +1,221 @@
+# =================
```
Comment (dhil) [inline generic/nbody/simple.links]: Move to `generic/` and rename to `nbody.links`.

**Review by dhil — COMMENTED** [review]
LGTM

**Comment by samo-novak** [comment]
I moved the files @dhil

**Review by dhil — APPROVED** [review]

**Pull request merged**